rust/lance/src/index/vector/ivf.rs

```diff
@@ -2417,6 +2418,7 @@ mod tests {
 
         let gt = ground_truth(&mat, query.values(), k);
         let recall = results.intersection(&gt).count() as f32 / k as f32;
+        println!("recall: {}", recall);
         assert!(
             recall >= 0.8,
```

Comment:
nit: could we add a random seed? then it's be easier to measure if recall is better in unit test.

Say `seed: 42` should get at least `0.88`, and we can build up this pass threshold everytime we make improvements. (probably also include a few more seeds)